azure-pipelines.release-fluentui.yml

```diff
@@ -1,74 +1,195 @@
+        condition: and(succeeded(), eq(variables.publishOfficial, true))
+        timeoutInMinutes: 2
+        inputs:
+          script: |
+            yarn logout
+
+            export NPM_TOKEN=$(npmToken)
+
+            echo "@fluentui:registry=http://registry.npmjs.org/" > .npmrc
+            echo "registry=http://registry.npmjs.org/" >> .npmrc
+            echo //registry.npmjs.org/:_authToken=$(npmToken) >> .npmrc
+
+            touch packages/fluentui/.npmrc
+
+            echo "@fluentui:registry=http://registry.npmjs.org/" > packages/fluentui/.npmrc
+            echo "registry=http://registry.npmjs.org/" >> packages/fluentui/.npmrc
+            echo //registry.npmjs.org/:_authToken=$(npmToken) >> packages/fluentui/.npmrc
+
+            yarn release:fluentui:$(version) --yes
+
+            yarn release:fluentui:post-validation
+
+      - task: DeleteFiles@1
+        inputs:
+          SourceFolder: $(Build.SourcesDirectory)
+          Contents: '**/*'
+        condition: always()
+
+  - job: Job_build_publish_doc
+    workspace:
+      clean: all
+    displayName: Build and Publish Docsite
+    dependsOn: Job_build_publish
+    # run this job when the previous job is succeeded or when publishDocsiteOnly is true
+    condition: or(succeeded(), eq(variables.publishDocsiteOnly, true))
+
+    steps:
+      - template: .devops/templates/tools.yml
+
+      - task: Bash@3
+        displayName: Yarn
+        inputs:
+          filePath: yarn-ci.sh
+
+      - task: CmdLine@2
+        displayName: 'Set base path and version'
+        condition: and(succeeded(), eq(variables.publishOfficial, true))
+        inputs:
+          script: |
+            ver=`node -p "require('./packages/fluentui/react-northstar/package.json').version"`
+            echo "Docsite base path published for version $ver"
+            echo "##vso[task.setvariable variable=deployBasePath]$ver"
+            echo "##vso[task.setvariable variable=docsiteVersion]$ver"
+            echo "##vso[task.setvariable variable=nightlyRelease]"
+
+      - task: CmdLine@2
+        displayName: Build
+        inputs:
+          script: |
+            echo deployBasePath $(deployBasePath) docsiteVersion $(docsiteVersion) nightlyRelease $(nightlyRelease)
+            NODE_ENV=production yarn build:fluentui:docs
+
+      - task: PublishPipelineArtifact@1
+        displayName: Publish Docsite as Pipeline Artifact
+        inputs:
+          path: packages/fluentui/docs/dist
+          artifactName: docsite_v$(docsiteVersion)
+
+      - task: AzureUpload@2
+        displayName: Upload to Azure
+        inputs:
+          SourcePath: packages/fluentui/docs/dist
+          azureSubscription: 'UI Fabric (private)'
+          storage: fluentsite
+          ContainerName: $web
+          BlobPrefix: $(deployBasePath)
+          Gzip: true
+
+      - task: DeleteFiles@1
+        inputs:
+          SourceFolder: $(Build.SourcesDirectory)
+          Contents: '**/*'
+        condition: always()
```

Comment:
Do I correctly get that we cannot reuse `.devops/templates/cleanup.yml` because it includes a check for changed files?

Reply:
Yes, we always have changed files, the packed tarballs, and lerna always adds some githead in package.json.

Reply:
Not sure about the githeads (I've seen that before and it doesn't happen every time either)

but for the packed tarballs, can't the pipeline pass a path not in the repo ? like $(Build.ArtifactsDirectory) ?

Reply:
So in the end I went with `Build.ArtifactStagingDirectory` for storing the tarballs. In azure doc it says this directory gets purged by each build.
And used clean script for both jobs